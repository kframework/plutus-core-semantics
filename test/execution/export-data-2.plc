(program

(module Nat (import) (export ((MyNat (S Z))) ())
  (data MyNat ()
        (S (con Nat.MyNat))
        (Z))
)

(module Foo (import Nat) (export () ())
  (define result
    (case (con Nat.S (con Nat.Z))
          (Nat.S (x) (con Nat.Z))
          (Nat.Z ()  -1))))

)
