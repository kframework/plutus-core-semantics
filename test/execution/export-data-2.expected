<T>
  <k> . </k>
  <lenv> .Map </lenv>
  <genv> ( Foo . result ) |-> ( con Nat . Z .ValList ) </genv>
  <modules>
    <module>
      <curr> 0 </curr>
      <name> Foo </name>
      <data> .TyExpList </data>
      <decl> .NameList </decl>
      <import> Nat  .ModNmList </import>
      <exportTy> .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </module>
    <module>
      <curr> 0 </curr>
      <name> Nat </name>
      <data> ( MyNat ( S  ( Z  .ConList ) ) )  .TyExpList </data>
      <decl> .NameList </decl>
      <import> .ModNmList </import>
      <exportTy> ( MyNat ( S  ( Z  .ConList ) ) )  .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </module>
  </modules>
</T>
